File: codegen/results/v002/qwen3.6_35b-a3b-coding-nvfp4_nothink/20260418_194110/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const filePath = path.join(process.cwd(), 'input', 'data.csv');
const content = fs.readFileSync(filePath, 'utf-8');
const lines = content.trim().split('\n');
const header = lines[0].split(',');

const result: Array<Record<string, string | number>> = [];

for (let i = 1; i < lines.length; i++) {
  const line = lines[i].trim();
  if (!line) continue;
  const values = line.split(',');
  const obj: Record<string, string | number> = {};
  obj['Name'] = values[0];
  obj['Age'] = parseInt(values[1], 10);
  obj['Email'] = values[2];
  obj['Score'] = parseFloat(values[3]);
  result.push(obj);
}

console.log(JSON.stringify(result));

File: codegen/results/v002/qwen3.6_35b-a3b-coding-nvfp4_nothink/20260418_194110/python/001_csv_to_json/solution.py
import csv
import json
import sys

def main():
    input_file = 'input/data.csv'
    output = []
    
    with open(input_file, 'r', newline='') as f:
        reader = csv.DictReader(f)
        for row in reader:
            obj = {
                "Name": row['Name'],
                "Age": int(row['Age']),
                "Email": row['Email'],
                "Score": float(row['Score'])
            }
            output.append(obj)
    
    print(json.dumps(output))

if __name__ == '__main__':
    main()